### PR TITLE
nds: add validation for `tls_skip_verify`

### PR DESCRIPTION
**Pull request by lgfa29**
`TLSSkipVerify` is ignored for Nomad service checks.

### CODE REVIEW

File: nomad/structs/services.go [hunk 1]
```diff
@@ -388,6 +388,11 @@ func (sc *ServiceCheck) validateNomad() error {
 		return fmt.Errorf("tls_server_name may only be set for Consul service checks")
 	}
 
+	// tls_skip_verify is consul only
+	if sc.TLSSkipVerify {
+		return fmt.Errorf("tls_skip_verify may only be set for Consul service checks")
```
Comment (jrasell) [inline nomad/structs/services.go:393]: The error isn't using formatting, so we could probably just use `errors.New`.

**Review by jrasell — APPROVED** [review]
LGTM, with a minor comment about the error generation.

**Review by shoenig — APPROVED** [review]
LGTM; just noting there is an issue for adding support for this in https://github.com/hashicorp/nomad/issues/16213

File: nomad/structs/services.go (hunk 1 above)
Comment (lgfa29) [reply nomad/structs/services.go]: Ops, copy-pasta. I adjusted the others as well in https://github.com/hashicorp/nomad/pull/18333/commits/37b9a2d4614b69bc0b58d4ac930201db782ebe98

**Pull request merged**